### PR TITLE
Removed deprecated methods

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
These methods was deprecated more than one year ago

## Core
- `Shutdown` (deprecated in 1.0.6), `TerminationTask` (deprecated in 1.0.6), `AwaitTermination` (deprecated in 1.0.6), `WaitForShutdown` (deprecated in 0.8) in `ActorSystem` class.
- [Become](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/UntypedActor.cs#L61) method from `UntypedActor` class. (deprecated in 1.0.0)
- [Become](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ReceiveActor.cs#L112) and [`Receive<T>(Func<T, Task>  handler)`](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ReceiveActor.cs#L134) methods from `ReceiveActor` class. (deprecated in 1.0.0)
- [Become](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ActorBase.cs#L227) and [Unbecome](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ActorBase.cs#L268) methods in `ActorBase` class (deprecated in 1.0.0)
- [UntypedActorWithBoundedStash](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/Stash/UntypedActorWithBoundedStash.cs#L17) class (deprecated in 1.0.0)
- [UntypedActorWithUnboundedStash](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/Stash/UntypedActorWithUnboundedStash.cs#L17) class (deprecated in 1.0.0)
- [IActorLogging](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ActorBase.cs#L82) - (deprecated in 0.8.0)
- [Become](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ActorCell.cs#L301) and [Unbecome](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Actor/ActorCell.cs#L310) methods in `ActorCell` class (deprecated in 1.0.0)
- [Warn](https://github.com/akkadotnet/akka.net/blob/d86eb6eb4fe512aac603cbc20bddc420acaec74a/src/core/Akka/Event/ILoggingAdapter.cs#L64) in ILoggingAdapter interface (deprecated in 1.0.2)
- [GetMillisDuration](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka/Configuration/Hocon/HoconValue.cs#L369) in HoconValue class  (deprecated in 0.6.4)

## Persistence
- [Become](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/PersistentActor.cs#L323) method from `PersistentActor` class.
- [`Persist<TEvent>(IEnumerable<TEvent> events, Action<TEvent> handler)`](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/Eventsourced.cs#L387) method (deprecated in 1.0.7)
- [`PersistAsync<TEvent>(IEnumerable<TEvent> events, Action<TEvent> handler)`](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/Eventsourced.cs#L454) method (deprecated in 1.0.7)
- [IWithPersistenceId](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/Persistent.cs#L21) class (deprecated in 1.0.7)
- [`Defer<TEvent>(TEvent evt, Action<TEvent> handler)`](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/Eventsourced.cs#L500) and [`Defer<TEvent>(IEnumerable<TEvent> events, Action<TEvent> handler)`](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/Eventsourced.cs#L512) methods (deprecated in 1.0.7)

## Cluster
- [GetMillisDurationWithOffSwitch](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Cluster/Util.cs#L111) method in Akka.Cluster.Util class (deprecated in 0.7.0)

## TestKit
- [Apply](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.TestKit/TestLatch.cs#L205) method in TestLatch class  (deprecated in 1.0.0)
- [Result](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.TestKit/TestLatch.cs#L191) method in TestLatch class  (deprecated in 1.0.0)
- [TestLatch.constructor](https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.TestKit/TestLatch.cs#L106) in TestLatch  (deprecated in 1.0.0)

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy conflict with the .approved file

**Comment by Danthar** [comment]
I see the deprecated scheduler extensions are still in here. Can we please get rid of those. Some things that are deprecated since 1.0.0 should just be binned imho. If not, how long are we going to keep them ?

File: src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt [hunk 1]
```diff
@@ -227,8 +227,6 @@ namespace Akka.Persistence
         public override void AroundPreStart() { }
         protected override bool AroundReceive(Akka.Actor.Receive receive, object message) { }
         [System.ObsoleteAttribute("Use DeferAsync instead")]
-        public void Defer<TEvent>(TEvent evt, System.Action<TEvent> handler) { }
-        [System.ObsoleteAttribute("Use DeferAsync instead")]
         public void Defer<TEvent>(System.Collections.Generic.IEnumerable<TEvent> events, System.Action<TEvent> handler) { }
```
Comment (Horusiath) [inline src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt]: Don't we want to remove this Defer as well?

File: src/examples/FSharp.Api/MapReduce.fs [hunk 2]
```diff
@@ -65,6 +65,6 @@ let main() =
         let! res = master <? Collect
         for (k, v) in res do
             printfn "%s\t%d" k v
-        system.Shutdown()
+        system.WhenTerminated.Wait()
```
Comment (Horusiath) [inline src/examples/FSharp.Api/MapReduce.fs]: I think, this is not an equivalent of `Shutdown()` ;)

File: src/core/Akka/Actor/ActorCell.Children.cs [hunk 3]
```diff
@@ -325,7 +325,7 @@ private bool TryGetChildRestartStatsByName(string name, out ChildRestartStats ch
         /// </summary>
         /// <param name="name">TBD</param>
         /// <returns>TBD</returns>
-        [Obsolete("Use TryGetSingleChild")]
+        [Obsolete("Use TryGetSingleChild [0.7.1]")]
         public IInternalActorRef GetSingleChild(string name)
```
Comment (Horusiath) [inline src/core/Akka/Actor/ActorCell.Children.cs]: Do we even care to keep it in public API? Maybe we should get rid of it?

File: src/core/Akka/Actor/ActorCell.cs [hunk 4]
```diff
@@ -230,7 +230,7 @@ public void Init(bool sendSupervise, MailboxType mailboxType)
         /// </summary>
         /// <param name="name">TBD</param>
         /// <returns>TBD</returns>
-        [Obsolete("Use TryGetChildStatsByName", true)]
+        [Obsolete("Use TryGetChildStatsByName [0.7.1]", true)]
         public IInternalActorRef GetChildByName(string name)   //TODO: Should return  Option[ChildStats]
```
Comment (Horusiath) [inline src/core/Akka/Actor/ActorCell.cs]: Same as with `GetSingleChild`

File: src/core/Akka/Actor/ActorCell.Children.cs (hunk 3 above)
Comment (alexvaluyskiy) [inline src/core/Akka/Actor/ActorCell.Children.cs]: Yes, we could make it internal

**Comment by alexvaluyskiy** [comment]
@Danthar Yes, we could remove Scheduler methods. But I did not understand, we can't use the same API as  Scala has, like this?
https://gist.github.com/alexvaluyskiy/e5c17836162631c3da3917ff200eb23f

Our API is overcomplicated, I'm talking about `ScheduleTellOnce`, `ScheduleTellOnceCancelable` instead of just one `ScheduleTellOnce`

File: src/examples/FSharp.Api/MapReduce.fs (hunk 2 above)
Comment (alexvaluyskiy) [inline src/examples/FSharp.Api/MapReduce.fs]: Should I change to `system.Terminate()`

File: src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt (hunk 1 above)
Comment (alexvaluyskiy) [inline src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt]: Will do

**Comment by Danthar** [comment]
@alexvaluyskiy I agree with that.

**Pull request merged**